haas/api.py

```diff
@@ -133,7 +133,65 @@ def project_detach_node(project, node):
     project.nodes.remove(node)
     db.session.commit()
 
-
+@rest_call('PUT', '/network/<network>/access/<project>')
+def network_grant_project_access(project, network):
+   """Add access to <network> to <project>.
+
+   If the project or network does not exist, a NotFoundError will be raised.
+   If the project already has access to the network a DuplicateError will be raised.
+   """
+   network = _must_find(model.Network, network)
+   project = _must_find(model.Project, project)
+   #must be admin or the owner of the network to add projects
+   get_auth_backend().require_project_access(network.owner)
+
+   if project in network.access:
+       raise DuplicateError('Network %s is already in project %s'%
+                            (network.label, project.label))
+
+   network.access.append(project)
+   db.session.commit()
+
+@rest_call('DELETE', '/network/<network>/access/<project>')
+def network_revoke_project_access(project, network):
+    """Remove access to <network> from <project>.
+
+    If the project or network does not exist, a NotFoundError will be raised.
+    If the project is the owner of the network a BlockedError will be raised.
+    """
+    network = _must_find(model.Network, network)
+    project = _must_find(model.Project, project)
+    #must be admin, the owner of the network, or <project> to remove <project>.
+    authorized = get_auth_backend().have_project_access(network.owner)
+    
+    if network.access:
+        for proj in network.access:
+            authorized = authorized or ((proj.label == project.label) and 
+                                        get_auth_backend().have_project_access(proj))
+            
+    if not authorized:
+        raise AuthorizationError("You are not authorized to remove the specified project form this network.")
+   
+    if project not in network.access:
+        raise NotFoundError("Network %r is not in project %r"%
+                            (network.label, project.label))
+
+    if project is network.owner:
+        raise BlockedError("Project %r is owner of network %r and its access cannot be removed"%
+                           (project.label, network.label))
+
+    for attachment in network.attachments:
```

Comment:
Im getting an error:  `AttributeError: Neither 'InstrumentedAttribute' object not 'Comparator' object associated with NetworkAttachment.nic had an attribute 'owner'`

this is the query: model.NetworkAttachment.query.join(model.Project, model.NetworkAttachment.nic.owner.project.label == Project.label).count()

I am a) not quite sure what this error means and b) unsure of how to actually make a comparison between an network attachment and a project without going through the owner of the nic